docs/DeveloperGuide.md

```diff
@@ -2,37 +2,189 @@
 
 ## Acknowledgements
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the
+original source as well}
 
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### Architecture
 
+The high-level design of TutorLink is as depicted in the following **Architecture Diagram**:
 
-## Product scope
-### Target user profile
+![Architecture.png](diagrams/Architecture.png)
```

Comment:
confusing diagram, do not understand what the arrows and lines are trying to say, maybe add a legend of what each line